### PR TITLE
(GH-2802) Document using '/bin/sh -c' for the Docker transport

### PR DESCRIPTION
**Pull request by beechtom**
This adds documentation to the Troubleshooting page that describes using
`/bin/sh -c` to run a command using the Docker transport when
interpolating environment variables into the command.

!no-release-note

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]

File: documentation/troubleshooting.md [hunk 1]
```diff
@@ -255,6 +255,48 @@ for your use case and security practices, you can manually decrypt the key by ru
...
+```
+
+This results in the expected output:
+
+```shell
+Started on docker://example...
+Finished on docker://example:
+  hello
+Successful on 1 target: docker://example
+Ran on 1 target in 0.59 sec
```
Comment (lucywyman) [inline documentation/troubleshooting.md:297]: I think this should include a note that you can configure Docker to always run with `/bin/sh -c` by setting `shell-command` for the transport, which is safe to do even if some commands already include `/bin/sh -c`.

Comment (beechtom) [reply documentation/troubleshooting.md:297]: Good idea. Added. :)

File: documentation/troubleshooting.md [hunk 2]
```diff
@@ -255,6 +255,66 @@ for your use case and security practices, you can manually decrypt the key by ru
...
+```
+
+You can configure the Docker transport to always execute a new shell process
+when running commands by configuring the `docker.shell-command` configuration
+option in your inventory file or `bolt-defaults.yaml` file:
```
Comment (hestonhoffman) [inline documentation/troubleshooting.md:302]:
```suggestion
You can configure the Docker transport to always execute a new shell process
when running commands by setting the `docker.shell-command` configuration
option in your inventory file or `bolt-defaults.yaml` file:
```
Just reducing the repetition of configure/configuring a bit :)

Comment (beechtom) [reply documentation/troubleshooting.md:302]: 😂  Good call.

**Review by lucywyman — APPROVED** [review]

**Pull request merged**